### PR TITLE
MongoDB vs. MySQL

### LINKED ISSUE #1112: MongoDB vs. MySQL
### Brief Summary: 
Describe the what, why, and how of your content idea in 2-5 sentences.

Database stores real-time information and handles a massive volume of queries at a fast rate. The volume of data keeps on increasing in recent years. Both MongoDB and MYSQL have large community users.
The article will be about analyzing key differences and similarities between the two databases. The article will highlight their differences in automation, structure, security, documentation, etcetera. It will also enumerate the pros and cons of using each.

### Key Takeaways: 
What are the 3-5 most important things the reader should understand or be able to do after reading this article?

1.	What is MongoDB? What is MYSQL?
2.	Applications of these databases, where are they deployed?
3.	What are the characteristics of both MongoDB and MYSQL?
4.	What are the differences between MongoDB and MYSQL? 
5.	What are the advantages and disadvantages of using MongoDB and MYSQL?

### References: 
Please list links to any published content/research that you intend to use to support/guide this article. (If none, please indicate 'N/A'.)

### PR DESCRIPTION
**Pull request by Pius1991**
This fixes #1112

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.

**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process and style-once published, you can then submit any number of articles.)**
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

File: articles/mongodb-vs-mysql/index.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+- It has been around for quite some time; hence publications and knowledge sources are many. It is a community-driven database.
+- It supports a password security system, and therefore it's somehow secure.
+- It can handle large unstructured data faster than MongoDB.
+
+#### Disadvantages of using MySQL.
+
+- It may take a lot of time and effort to perform some tasks, even for some tasks requiring an automatic update, such as incremental back-ups.
+- It is not scalable. It is not possible to use replica sets in MySQL.
+- It has no in-built [XML](https://en.wikipedia.org/wiki/XML#:~:text=Extensible%20Markup%20Language%20(XML)%20is,free%20open%20standards%E2%80%94define%20XML) (Extensible Markup Language) and [OLAP](https://olap.com/olap-definition/) (Online Analytical Processing).
+
```
Comment (lalith1403) [inline articles/mongodb-vs-mysql/index.md:70]: Can you take a practical point of view? Compare the two with respect to two projects that you may have worked on. Giving a real-world example helps the reader understand the pros and cons better. The pros and cons that you mention are good. But, I wouldn't be interested in them until I know their applications.

File: articles/mongodb-vs-mysql/index.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+| [Schemas](https://www.tutorialspoint.com/dbms/dbms_data_schemas.htm) (Skelton structure that represents the logic of the database). | It has a strict schema. Before storing anything, one needs to define tables and columns clearly. Also, ensure that every row in the table should have the same column. | It has a dynamic schema. Several documents can be dropped with no relation, only that documents need to be supported by data structures. |
+| Attack risks | It is prone to [SQL injection attacks](https://www.imperva.com/learn/application-security/sql-injection-sqli/) (a vector that uses malicious SQL code). | Less risk of attack. Its design makes it hard to attack. |
+| Organized By | Data is organized by a table, row, and columns. | Data is organized by collection, document, and field classification. |
+| Performance and Speed | It is better for smaller databases and more general solutions. | It is better for handling massive, unstructured data sets as it has in-memory speed. |
+| Website | Mysql.com | Mongodb.com |
```
Comment (lalith1403) [inline articles/mongodb-vs-mysql/index.md:37]: I think this feature is too generic.

Comment (lalith1403) [reply articles/mongodb-vs-mysql/index.md:37]: and redundant.

File: articles/mongodb-vs-mysql/index.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+| Developer | Oracle corporation does the ongoing development. | MongoDB, Inc. does the ongoing development. |
+| Scaling | Allows using more powerful database servers (vertical scaling). | Allows adding more database servers (horizontal scaling). |
+| Support provider | It offers [Oracle lifetime support](https://www.mysql.com/support/), though at different versions. Each level offers a knowledge base, maintenance releases, bug fixes, patches, and updates. | Offers Enterprise-Grade Support. It allows flexibility to upgrade to the newer versions. It has unlimited access to security fixes, updates, and many more. |
+| Community of users | It has a large community of users, as it started several years ago. | It has a relatively smaller community than that of MySQL, as it is newer. |
+| Command speed | MySQL is slower than MongoDB in executing the same commands like select, update, and insert. | MongoDB is faster in the execution of commands than MySQL. |
```
Comment (lalith1403) [inline articles/mongodb-vs-mysql/index.md:42]: Saying A is greater than B, and B is lesser than A is the same thing. Avoid redundant comparisons. I want to know which aspect of c++ makes it slower than MongoDB? 

In the above table, avoid such redundancies. Wherever applicable, remove it or base the arguments with facts.

**Review by lalith1403 — CHANGES_REQUESTED** [review]
When a person reads your article, expect them to know nothing in the beginning and know everything in your topic towards the end. You explain concepts well but don't go in-depth. Add some more additional material and let me know. 

Any statement that you make, base it on facts and reasoning.

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by hectorkambow** [comment]
@Pius1991  

Please take notes on the suggestions made by the peer reviewer - this article is a great article topic - but the information Im coming across seems very basic and lacking substance. 

The conclusion seemed to barely wrap any thing up. 

I found myself reworking and rewording sentences to try and add some substance.

**Comment by hectorkambow** [comment]
Please consider adding some more substantial information or examples to add some more value to the article and the community as a whole. 

The article was a bit rough around the edges - and it seemed like most information could be easily found elsewhere - we want to make sure we are putting together articles that solve major pain points with substance.

**Comment by Pius1991** [comment]
@hectorkambow  it's well noted, please re-open so that I can work on it.

**Comment by hectorkambow** [comment]
@Pius1991 I'll be closing this PR just to make room in the queue. 

You can re-submit the "revised and improved" article in a new PR (by transferring the files over) and link this to it for us to know 👍 
Thank you.

**Pull request closed without merging**